packages/rocketchat-slackbridge/slack.js

```diff
@@ -0,0 +1,1084 @@
+					}
+				} else {
+					topic = data.purpose.topic;
+					topic_creator = data.purpose.creator;
+				}
+			}
+
+			if (topic) {
+				const creator = this.rocket.findUser(topic_creator) || this.rocket.addUser(topic_creator);
+				logger.slack.debug('Setting room topic', rid, topic, creator.username);
+				RocketChat.saveRoomTopic(rid, topic, creator, false);
+			}
+		}
+	}
+
+	copyPins(rid, channelMap) {
+		const response = HTTP.get('https://slack.com/api/pins.list', { params: { token: this.apiToken, channel: channelMap.id } });
+		if (response && response.data && _.isArray(response.data.items) && response.data.items.length > 0) {
+			for (const pin of response.data.items) {
+				if (pin.message) {
+					const user = this.rocket.findUser(pin.message.user);
+					const msgObj = {
+						rid: rid,
+						t: 'message_pinned',
+						msg: '',
+						u: {
+							_id: user._id,
+							username: user.username
+						},
+						attachments: [{
+							'text' : this.rocket.convertSlackMsgTxtToRocketTxtFormat(pin.message.text),
+							'author_name' : user.username,
+							'author_icon' : getAvatarUrlFromUsername(user.username),
+							'ts' : new Date(parseInt(pin.message.ts.split('.')[0]) * 1000)
+						}]
+					};
+
+					RocketChat.models.Messages.setPinnedByIdAndUserId(`slack-${pin.channel}-${pin.message.ts.replace(/\./g, '-')}`, msgObj.u, true, new Date(parseInt(pin.message.ts.split('.')[0]) * 1000));
+				}
+			}
+		}
+	}
+
+	importMessages(rid, callback) {
+		logger.slack.info('importMessages: ', rid);
+		const rocketchat_room = RocketChat.models.Rooms.findOneById(rid);
+		if (rocketchat_room) {
+			if (this.getSlackChannel(rid)) {
+				this.copyChannelInfo(rid, this.getSlackChannel(rid));
+
+				logger.slack.debug('Importing messages from Slack to Rocket.Chat', this.getSlackChannel(rid), rid);
+				let results = this.importFromHistory(this.getSlackChannel(rid).family, { channel: this.getSlackChannel(rid).id, oldest: 1 });
+				while (results && results.has_more) {
+					results = this.importFromHistory(this.getSlackChannel(rid).family, { channel: this.getSlackChannel(rid).id, oldest: results.ts });
+				}
+
+				logger.slack.debug('Pinning Slack channel messages to Rocket.Chat', this.getSlackChannel(rid), rid);
+				this.copyPins(rid, this.getSlackChannel(rid));
+
+				return callback();
+			} else {
+				const slack_room = this.postFindChannel(rocketchat_room.name);
+				if (slack_room) {
+					this.addSlackChannel(rid, slack_room.id);
+					return this.importMessages(rid, callback);
+				} else {
+					logger.slack.error('Could not find Slack room with specified name', rocketchat_room.name);
+					return callback(new Meteor.Error('error-slack-room-not-found', 'Could not find Slack room with specified name'));
+				}
+			}
+		} else {
+			logger.slack.error('Could not find Rocket.Chat room with specified id', rid);
+			return callback(new Meteor.Error('error-invalid-room', 'Invalid room'));
+		}
+	}
+
+}
+
```

Comment:
This seemed the most appropriate way to resolve 'require/dependency' so the two JS files could see each other